scripts/globals/mobskills/charm_copycat.lua

```diff
@@ -0,0 +1,51 @@
+-----------------------------------
+-- Charm Copycat
+-----------------------------------
+require("scripts/globals/mobskills")
+require("scripts/globals/settings")
+require("scripts/globals/status")
+require("scripts/globals/msg")
+-----------------------------------
+local mobskill_object = {}
+
+mobskill_object.onMobSkillCheck = function(target, mob, skill)
+    return 0
+end
+
+mobskill_object.onMobWeaponSkill = function(target, mob, skill)
+    local typeEffect = xi.effect.CHARM_I
+    local power = 0
+    local job = target:getMainJob()
+    local pos = mob:getPos()
+
+    if not target:isPC() then
+        skill:setMsg(xi.msg.basic.SKILL_MISS)
+        return typeEffect
+    end
+
+    local msg = xi.mobskills.mobStatusEffectMove(mob, target, typeEffect, power, 3, 30)
+
+    if msg == xi.msg.basic.SKILL_ENFEEB_IS then
+        mob:charm(target)
+    end
+
+    if job == 9 then -- BST
+        GetMobByID(mob:getID()+2):setSpawn(pos.x, pos.y, pos.z, pos.rot)
```

Comment:
`mob:getID()` can be cast to a variable so that you call it only once, reducing the number of times the code jumps from lua to core and back.